### PR TITLE
#25 add job service and repository

### LINKED ISSUE #25: add Job service and repository

### PR DESCRIPTION
**Pull request by student-google**
this closes #25

### CODE REVIEW

**Comment by student-google** [comment]
#25 this closed

File: JobSeek/JobSeek.Api/Controllers/JobCategoryController.cs [hunk 1]
```diff
@@ -36,6 +36,7 @@ public JobCategory Create(JobCategoryInput input)
             return service.Create(input);
         }
 
+        //deleted api
```
Comment (zoha-shobbar) [inline JobSeek/JobSeek.Api/Controllers/JobCategoryController.cs:39]: remove this line

File: JobSeek/JobSeek.Api/Controllers/JobsController.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class JobsController : IJobSerivce
+    {
+        private readonly IJobSerivce _jobSerivce;
+
+        public JobsController(IJobSerivce jobSerivce)
+        {
+            this._jobSerivce = jobSerivce;
```
Comment (zoha-shobbar) [inline JobSeek/JobSeek.Api/Controllers/JobsController.cs]: remove "this" keyword

File: JobSeek/JobSeek.Api/Repository/Contracts/IJobRepository.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿using JobSeek.Api.Models.Entities;
+using JobSeek.Api.Models.Input;
+
+namespace JobSeek.Api.Repository.Contracts
+{
+    public interface IJobRepository
+    {
+        List<Job> GetAll();
+        Job GetById(int id);
+        Job create(Job job);
```
Comment (zoha-shobbar) [inline JobSeek/JobSeek.Api/Repository/Contracts/IJobRepository.cs]: change create to Create

File: JobSeek/JobSeek.Api/Repository/Contracts/IJobRepository.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+        bool delete(int id);
```
Comment (zoha-shobbar) [inline JobSeek/JobSeek.Api/Repository/Contracts/IJobRepository.cs]: change delete to Delete

File: JobSeek/JobSeek.Api/Repository/JobRepository.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+namespace JobSeek.Api.Repository
+{
+    public class JobRepository : IJobRepository
+    {
+        private readonly DataContext _dataContext;
+
+        public JobRepository(DataContext dataContext) 
+        {
+            this._dataContext = dataContext;
```
Comment (zoha-shobbar) [inline JobSeek/JobSeek.Api/Repository/JobRepository.cs]: remove "this" keyword

File: JobSeek/JobSeek.Api/Repository/JobRepository.cs [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        }
+        public Job create(Job job)
```
Comment (zoha-shobbar) [inline JobSeek/JobSeek.Api/Repository/JobRepository.cs]: change create to Create

File: JobSeek/JobSeek.Api/Repository/JobRepository.cs [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        {
+            _dataContext.Jobs.Add(job);
+            _dataContext.SaveChanges();
+            return job;
+
+        }
+
+        public bool delete(int id)
```
Comment (zoha-shobbar) [inline JobSeek/JobSeek.Api/Repository/JobRepository.cs]: change delete to Delete

File: JobSeek/JobSeek.Api/Services/Contracts/IJobSerivce.cs [hunk 8]
```diff
@@ -0,0 +1,13 @@
+﻿using JobSeek.Api.Models.Entities;
+using JobSeek.Api.Models.Input;
+
+namespace JobSeek.Api.Services.Contracts
+{
+    public interface IJobSerivce
+    {
+        List<Job> GetAll();
+        Job GetById(int id);
+        Job create(Job job);
```
Comment (zoha-shobbar) [inline JobSeek/JobSeek.Api/Services/Contracts/IJobSerivce.cs]: change create to Create

File: JobSeek/JobSeek.Api/Services/Contracts/IJobSerivce.cs [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+        bool delete(int id);
```
Comment (zoha-shobbar) [inline JobSeek/JobSeek.Api/Services/Contracts/IJobSerivce.cs]: change delete to Delete

File: JobSeek/JobSeek.Api/Services/Implementation/JobService.cs [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
+namespace JobSeek.Api.Services.Implementation
+{
+    public class JobService : IJobSerivce
+    {
+        private readonly IJobRepository _repository;
+
+        public JobService(IJobRepository repository) 
+        {
+            this._repository = repository;
```
Comment (zoha-shobbar) [inline JobSeek/JobSeek.Api/Services/Implementation/JobService.cs]: remove "this" keyword

File: JobSeek/JobSeek.Api/Services/Implementation/JobService.cs [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+        {
+            return _repository.GetAll();
+        }
+
+        public Job GetById(int id)
+        {
+            return _repository.GetById(id);
+        }
+        
+        public Job create(Job job)
```
Comment (zoha-shobbar) [inline JobSeek/JobSeek.Api/Services/Implementation/JobService.cs]: in this method should take a JobInput instead of Job as a parameter

File: JobSeek/JobSeek.Api/Services/Implementation/JobService.cs [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+        {
+            return _repository.create(job);
+        }
+
+        public bool delete(int id)
+        {
+            var ExistEntity = _repository.delete(id);
+            if (id == null)
+                throw new Exception("this is not found!");
+            return ExistEntity;
```
Comment (zoha-shobbar) [inline JobSeek/JobSeek.Api/Services/Implementation/JobService.cs]: why you return the existEntity? why don't call the delete in the repository?

File: JobSeek/JobSeek.Api/Services/Contracts/IJobSerivce.cs (hunk 8 above)
Comment (zoha-shobbar) [inline JobSeek/JobSeek.Api/Services/Contracts/IJobSerivce.cs]: in this method should take a JobInput instead of Job as a parameter

File: JobSeek/JobSeek.Api/Services/Contracts/IJobSerivce.cs [hunk 13]
```diff
@@ -0,0 +1,13 @@
+﻿using JobSeek.Api.Models.Entities;
+using JobSeek.Api.Models.Input;
+
+namespace JobSeek.Api.Services.Contracts
+{
+    public interface IJobSerivce
```
Comment (zoha-shobbar) [inline JobSeek/JobSeek.Api/Services/Contracts/IJobSerivce.cs:6]: you need an update method

File: JobSeek/JobSeek.Api/appsettings.json [hunk 14]
```diff
@@ -7,6 +7,6 @@
   },
   "AllowedHosts": "*",
   "ConnectionStrings": {
-    "DefaultConnection": "Data Source=ZOHAPC\\ZSHO_SRVR19_01;Initial Catalog=JobSeek;Integrated Security=True;Pooling=False;Encrypt=False"
```
Comment (zoha-shobbar) [inline JobSeek/JobSeek.Api/appsettings.json:10]: dont remove this line, just comment it

File: JobSeek/JobSeek.Api/Controllers/JobCategoryController.cs [hunk 15]
```diff
@@ -33,9 +33,10 @@ public JobCategory Get(int id)
         [HttpPost]
         public JobCategory Create(JobCategoryInput input)
         {
-            return service.Create(input);
```
Comment (zoha-shobbar) [inline JobSeek/JobSeek.Api/Controllers/JobCategoryController.cs:36]: why did you remove this line?

**Review by zoha-shobbar — APPROVED** [review]
